### DISCUSSION TITLE
Move the wiki contents to the repo

### DISCUSSION THREAD

**Discussion by balacij** [JacquesCarette/Drasil#3745] [Ideas]
Rather than a new feature, this is more for the development process of the wiki.

GitHub Wiki works by creating a separate git repository (repo) for each GitHub-hosted repo. For example, with mainline Drasil, we have `git@github.com:JacquesCarette/Drasil.wiki.git` (a separate repo) hosting all the wiki contents. This scheme is common for all GitHub-hosted repos.

I have two minor gripes with the way they handle the separated wiki repos:
1.  we can't (easily) subscribe to notifications to changes on the wiki pages (similar to tracking PRs), and
2. changes to wiki pages don't go through a review process.

On (2), I think it would be great to be able to review wiki change proposals in a style similar to pull requests on GitHub.

**Proposal**:

My immediate thought is that we can move the contents of the wiki to our main repo (say, in a `wiki/` folder) so that we can use GitHub's PR system to review and track changes to the wiki (and trace changes to Issues!). The CI can then be responsible for finally pushing accepted changes to the wiki. Thankfully, it seems that at least two projects ([deploy-wiki](https://github.com/actions4gh/deploy-wiki) and [actions-wiki](https://github.com/spenserblack/actions-wiki)) can do this stuff for us through reusable GitHub workflows as part of our CI.

**Considerations**:

I don't think the time commitment is much, I'm not sure if others have the same gripes with GitHub wikis either. Is it worth the time?

One side effect of this approach is that contributions to the wiki will be included in our main "contribution stats" pages. I don't have any issue with that -- it sounds like a good thing to me.

tldr:
* GitHub wikis are not tightly integrated with projects. We can't put changes through a similar review process as PRs.
* To mitigate, we can move the wiki contents to the main project code, use the same PR system we normally use, and then have the CI push changes.

**Comment by smiths** [comment]
I agree with this proposal.  I've had the same concerns about the wiki.  It is most notable when we have an issue created to make a change to the wiki.  It is difficult for the person assigned the tasks to easily show what they changed, and difficult to review the changes.  The structure of the wiki suggests that the GitHub infrastructure was made with the common bias against documentation.

Including wiki contributions in the "contributions stats" makes complete sense to me.  We don't share the above bias that documentation is somehow "less than" code.  In fact, good documentation can have a greater impact on [long-term productivity](https://arxiv.org/abs/2112.12580) than writing code.  :smile:

**Reply by balacij** [reply]
I'm glad that we're on the same page about it :smile: Indeed, I don't think it's being used as nearly as well as it could be.

I could not have said it better myself.

**Reply by JacquesCarette** [reply]
I agree as well.

**Comment by balacij** [comment]
Closed in https://github.com/JacquesCarette/Drasil/pull/3771
